orthanc/orthanc-raw/plugin/pixl.py

```diff
@@ -58,14 +58,27 @@ def OnChange(changeType, level, resourceId):  # noqa: ARG001
     This routes any stable study to a modality named PIXL-Anon if
     should_auto_route returns true
     """
-    if changeType == orthanc.ChangeType.STABLE_STUDY and should_auto_route():
+    if changeType != orthanc.ChangeType.STABLE_STUDY:
+        return
+    if _get_project_name_from_study_id(resourceId) == DICOM_TAG_PROJECT_NAME.PLACEHOLDER_VALUE:
+        logger.warning("Study {} has not been set with a pixl project name, deleting", resourceId)
+        orthanc.RestApiDelete(f"/studies/{resourceId}")
+        return
+    # delete via rest API and don't autoroute
```

Comment:
```suggestion
    # delete via rest API and don't autoroute
    if _get_project_name_from_study_id(resourceId) == DICOM_TAG_PROJECT_NAME.PLACEHOLDER_VALUE:
        logger.warning("Study {} has not been set with a pixl project name, deleting", resourceId)
        orthanc.RestApiDelete(f"/studies/{resourceId}")
        return
```

Shouldn't the comment go before the relevant code block?

Reply:
it really should, though I should have just deleted the comment because it doesn't tell you why, just reiterates what the code is telling you anyway